project-base/storefront/components/Layout/Header/AutocompleteSearch/AutocompleteSearchPopup.tsx

```diff
@@ -0,0 +1,268 @@
+                                                    GtmSectionType.category,
+                                                    category.name,
+                                                )
+                                            }
+                                        >
+                                            {category.name}
+                                        </SearchResultLink>
+                                    </li>
+                                ),
+                        )}
+                    </SearchResultSectionGroup>
+                </SearchResultSection>
+            )}
+
+            {articlesSearch.length > 0 && (
+                <SearchResultSection>
+                    <SearchResultSectionTitle>{`${t('Articles')} (${articlesSearch.length})`}</SearchResultSectionTitle>
+
+                    <SearchResultSectionGroup dataTestId={TEST_IDENTIFIER + '-articles'}>
+                        {articlesSearch.map(
+                            (article, index) =>
+                                index < AUTOCOMPLETE_ARTICLE_LIMIT && (
+                                    <li key={article.slug} data-testid={TEST_IDENTIFIER + '-articles-' + index}>
+                                        <SearchResultLink
+                                            href={article.slug}
+                                            type={article.__typename === 'ArticleSite' ? 'article' : 'blogArticle'}
+                                            onClick={() =>
+                                                onGtmAutocompleteResultClickEventHandler(
+                                                    autocompleteSearchQueryValue,
+                                                    GtmSectionType.article,
+                                                    article.name,
+                                                )
+                                            }
+                                        >
+                                            {article.name}
+                                        </SearchResultLink>
+                                    </li>
+                                ),
+                        )}
+                    </SearchResultSectionGroup>
+                </SearchResultSection>
+            )}
+
+            <div className="flex justify-center">
+                <Button
+                    size="small"
+                    onClick={() =>
+                        router.push({
+                            pathname: searchUrl,
+                            query: { q: autocompleteSearchQueryValue },
+                        })
+                    }
+                    dataTestId={TEST_IDENTIFIER + '-all-button'}
+                >
+                    {t('View all results')}
+                </Button>
+            </div>
+        </>
+    );
+};
+
+const SearchResultSection: FC = ({ children }) => {
+    return <div>{children}</div>;
+};
```

Comment:
Not sureif this is necessary?

Reply:
Yup, nothing has left here after refactoring.